packages/components/src/unit-control/index.tsx

```diff
@@ -198,7 +197,7 @@ function UnitControl(
 			}
 		}
 
-		return state;
+		return stateReducer?.( state, action ) ?? state;
```

Comment:
Because the default reducer is removed on L41 optional-chaining and nullish coalescence are used here. I like this style but didn't do the same in `NumberControl` because I wanted to see what others may prefer.